zerver/views/realm.py

```diff
@@ -79,16 +80,23 @@ def update_realm(request, user_profile, name=REQ(validator=check_string, default
         data['authentication_methods'] = authentication_methods
     # The message_editing settings are coupled to each other, and thus don't fit
     # into the do_set_realm_property framework.
-    if (allow_message_editing is not None and realm.allow_message_editing != allow_message_editing) or \
-       (message_content_edit_limit_seconds is not None and
-            realm.message_content_edit_limit_seconds != message_content_edit_limit_seconds):
+    if ((allow_message_editing is not None and realm.allow_message_editing != allow_message_editing) or
+        (message_content_edit_limit_seconds is not None and
+            realm.message_content_edit_limit_seconds != message_content_edit_limit_seconds) or
+        (edit_topics_by_admins_only is not None and
+            realm.edit_topics_by_admins_only != edit_topics_by_admins_only)):
```

Comment:
So, the more I look at this bit of code, and especially at the messages to the user, the more I think that it'd be better to separate these settings at the events level, and just use your handy `do_set_realm_property` mechanism for each of these settings separately, just like we do for most settings.

Some of these settings won't have any effect as long as `allow_message_editing` is False, but I think that's fine. Already `invite_required` and `invite_by_admins_only` work the same way.

This code will get a lot simpler that way, and I think the messages will too.